sigstore/_verify.py

```diff
@@ -209,7 +210,11 @@ def verify(
         if expected_cert_email is not None:
             # Check that SubjectAlternativeName contains signer identity
             san_ext = cert.extensions.get_extension_for_class(SubjectAlternativeName)
-            if expected_cert_email not in san_ext.value.get_values_for_type(RFC822Name):
+            if expected_cert_email not in san_ext.value.get_values_for_type(
+                RFC822Name
+            ) and expected_cert_email not in san_ext.value.get_values_for_type(
+                UniformResourceIdentifier
+            ):
```

Comment:
I think we're going to want a different or more generic name for this.

Per #155, I think the current plan is to make this into `--cert-identity` (and thus `expected_cert_identity`).